app/screens/post_options/index.ts

```diff
@@ -132,7 +132,7 @@ const enhanced = withObservables([], ({combinedPost, post, showAddReaction, loca
     );
 
     const canMarkAsUnread = combineLatest([currentUser, channelIsArchived]).pipe(
-        switchMap(([user, isArchived]) => of$(!isArchived && user?.id !== post.userId && !isSystemMessage(post))),
+        switchMap(([user, isArchived]) => of$((!isArchived && user?.id !== post.userId && !isSystemMessage(post)) || isFromWebhook(post))),
```

Comment:
This would allow to mark post from webhooks on archived channels as unread. I think the parenthesis should go a bit further to the right (after `!isArchived`).